db/src/Pos/DB/Rocks/Functions.hs

```diff
@@ -91,6 +93,12 @@ openNodeDBs recreate fp = do
         , lrcPath
         , miscPath
         ]
+
+    when (System.Info.os == "darwin") $ do
+        -- Prevent indexing of blocks on OSX
+        _ <- openFile (fp </> ".metadata_never_index") AppendMode
+        pure ()
```

Comment:
Why not just `writeFile` ? Or if you want to do `openFile`, at least we should `hClose` too.

Reply:
Or
```
withFile (fp </> ".metadata_never_index") AppendMode (\_ -> return ())
```

Reply:
also, what about
```
import Distribution.System

when (buildOS == OSX) $ do
```
sum types are less likely to have mistakes then string compares

Reply:
That's true, and that's why Cabal implements that interface, but we can't do the same without depending on Cabal.

Reply:
we already depend on Cabal for the SystemTag logic in the update system and in a dozen places for the current version from the cabal file